tests/test_v5_client.py

```diff
@@ -103,3 +105,134 @@ def test_failing_requests(requests_mock):
 
     with pytest.raises(LuminosoServerError):
         client.get('fail')
+
+
+# The logic in wait_for_build() and wait_for_sentiment_build() gets a little
+# complex, so we test that logic more thoroughly here.
+
+def _last_build_infos_to_mock_returns(last_builds):
+    """
+    Helper function for testing waiting for a build.  Turns a series of
+    last_build_info dictionaries into a list suitable for returning
+    sequentially from requests_mock.
+    """
+    return [{'json': {'last_build_info': build_info}}
+            for build_info in last_builds]
+
+
+def test_wait_for_build(requests_mock):
+    project_url = BASE_URL + 'projects/pr123456/'
+    client = LuminosoClient.connect(project_url, token='fake')
+
+    # A somewhat pared-down representation of what a project record's
+    # `last_build_info` field looks like in various states
+    build_running = {'start_time': 1590000000.0, 'stop_time': None,
+                     'sentiment': {}}
+    build_failed = {'start_time': 1590000000.0, 'stop_time': 1590000001.0,
+                    'sentiment': {}, 'success': False}
+    build_succeeded = {'start_time': 1590000000.0, 'stop_time': 1590000001.0,
+                       'sentiment': {}, 'success': True}
+
+    # If there is no build: error
+    requests_mock.get(project_url, json={'last_build_info': {}})
+    with pytest.raises(ValueError, match='not building'):
+        client.wait_for_build()
+
+    # If the build succeeds: the project's last build info
+    requests_mock.get(
+        project_url,
+        _last_build_infos_to_mock_returns(
```

Comment:
I'm confused about why we pass this.

Reply:
requests-mock lets you specify a list of returns to use in sequential calls.  Lines 142-147 are saying "Tell the mock request that the responses to a `get` call to `project_url` are the following list....".  It's directly analogous to line 137 above, which says "the response to a `get` call to `project_url` is a JSON-encoded `{'last_build_info': {}}`".

Reply:
Oh, I see -- `client.wait_for_build()` is hitting that endpoint multiple times.  I wasn't seeing where we were calling it more than once.